### PR TITLE
[SYCL] Optimize operator[] method of host accessor

### PR DESCRIPTION
**Pull request by romanovvlad**
Compiler cannot optimize out calls to getPtr(), getOffset() and getMemoryRange(), that were used in accessor::operator[], because they may have side effects(e.g. printf). The patch adds a member in the accessor which points to the actual data stored in the accessor_impl. This pointer is initialized once during construction and used instead of functions mentioned above.

### CODE REVIEW

**Review by cperkinsintel — APPROVED** [review]
LGTM

File: sycl/test/basic_tests/accessor/host_acc_opt.cpp [hunk 1]
```diff
@@ -0,0 +1,26 @@
+// RUN: %clangxx -O2 -fsycl -S -emit-llvm %s -o - | FileCheck %s
```
Comment (bader) [inline sycl/test/basic_tests/accessor/host_acc_opt.cpp:1]: I suppose this test is supposed to call only a host compiler.

Comment (romanovvlad) [reply sycl/test/basic_tests/accessor/host_acc_opt.cpp]: It there a convenient way to call the host compiler only?

Comment (bader) [reply sycl/test/basic_tests/accessor/host_acc_opt.cpp]: I think `-fsycl-host-only` might be helpful.

Comment (bader) [reply sycl/test/basic_tests/accessor/host_acc_opt.cpp]: https://github.com/intel/llvm/blob/sycl/sycl/test/lit.cfg.py#L89 - it looks like it's a front-end option.

**Comment by pvchupin** [comment]
@romanovvlad, post commit issues: https://github.com/intel/llvm/actions/runs/3113733433/jobs/5048727706

**Comment by romanovvlad** [comment]
> @romanovvlad, post commit issues: https://github.com/intel/llvm/actions/runs/3113733433/jobs/5048727706

Fix for them https://github.com/intel/llvm/pull/6875

**Pull request merged**